taxi/tests/test_views.py

```diff
@@ -0,0 +1,218 @@
+        self.assertRedirects(res, reverse("taxi:manufacturer-list"))
+
+    def test_delete_manufacturer(self):
+        manufacturer = Manufacturer.objects.create(
+            name="test name",
+            country="test country"
+        )
+        url = reverse("taxi:manufacturer-delete", args=[manufacturer.id])
+        res = self.client.post(url)
+        self.assertRedirects(res, reverse("taxi:manufacturer-list"))
+
+        with self.assertRaises(Manufacturer.DoesNotExist):
+            Manufacturer.objects.get(id=manufacturer.id)
+
+    def test_search_manufacturer(self):
+        Manufacturer.objects.create(name="1_test_manufacturer")
+        Manufacturer.objects.create(name="2_test_manufacturer")
+        res = self.client.get(
+            MANUFACTURER_URL, {"name": "1_test_manufacturer"}
+        )
+        self.assertEqual(res.status_code, 200)
+        manufacturers = Manufacturer.objects.filter(
+            name__icontains="1_test_manufacturer"
+        )
+        self.assertEqual(
+            list(res.context["manufacturer_list"]),
+            list(manufacturers),
+        )
+
+
+class PublicCarTest(TestCase):
+    def test_loggin_required_car_list(self):
+        res = self.client.get(CAR_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+    def test_loggin_required_car_detail(self):
+        res = self.client.get(CAR_DETAIL_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateCarTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="test",
+            password="12345",
+            license_number="TTT12345"
+        )
+        self.client.force_login(self.user)
+
+    def test_loggin_required_car_list(self):
+        manufacturer = Manufacturer.objects.create(
+            name="test name",
+            country="test country"
+        )
+        Car.objects.create(model="test model 1", manufacturer=manufacturer)
+        Car.objects.create(model="test model 2", manufacturer=manufacturer)
+        res = self.client.get(CAR_URL)
+        self.assertEqual(res.status_code, 200)
+        cars = Car.objects.all()
+        self.assertEqual(
+            list(res.context["car_list"]),
+            list(cars)
+        )
+        self.assertTemplateUsed("taxi:car-list")
+
+    def test_create_car(self):
+        manufacturer = Manufacturer.objects.create(
+            name="test name",
+            country="test country"
+        )
+        data = {
+            "model": "test_model",
+            "manufacturer": manufacturer.id,
+            "drivers": [self.user.id]
+        }
+        self.client.post(reverse("taxi:car-create"), data=data)
+
+        car = Car.objects.get(model="test_model")
+        self.assertEqual(car.model, data["model"])
+        self.assertEqual(car.manufacturer, manufacturer)
+
+
+class PublicDriverTest(TestCase):
+    def test_loggin_required_car_list(self):

```

Comment:
There's a typo in the method name `test_loggin_required_car_list`. It should be `test_login_required_driver_list` for clarity and correctness.